libsolutil/StringUtils.cpp

```diff
@@ -113,3 +115,47 @@ string solidity::util::suffixedVariableNameList(string const& _baseName, size_t
 	}
 	return result;
 }
+
+string solidity::util::decodeURI(string const& _uri)
+{
+	string decodedString;
+	for (size_t i = 0; i < _uri.size(); ++i)
+	{
+		char const ch = _uri.at(i);
```

Comment:
Why are you using `at` access here, since I don't see a way for this to throw an out of bounds exception?

Reply:
@nikola-matic me neither. But if I remember correctly, I was corrected to prefer .at() as good practice. I check the source code for references. Otherwise, yes, you're right. I think it's actually obvious here in this case that this can't go OOB.

Reply:
I don't actually mind the usage of `at` so much, as it is definitely safer, plus in the case of an eventual failure, it's definitely easier to debug an out of bands error vs a segfault, it's just that you're already doing the bounds checking in `if` and `for` here, so it's kinda unnecessary.

Reply:
I usually use it like an assert. So in contexts where I'd not feel the need to assert or I have already asserted I use `[]`, otherwise it's `at()`.